collector.go

```diff
@@ -277,6 +288,47 @@ func collectPerSubscriptionMetric(stats *stats, desc *prometheus.Desc, collectFu
 	})
 }
 
+func collectParkedMessagesPerSubscriptionMetric(stats *stats, desc *prometheus.Desc, ch chan<- prometheus.Metric) {
+	jp.ArrayEach(stats.subscriptionsStats, func(jsonValue []byte, dataType jp.ValueType, offset int, err error) {
+		eventStreamID, _ := jp.GetString(jsonValue, "eventStreamId")
+		groupName, _ := jp.GetString(jsonValue, "groupName")
+
+		// get parked messages max event number
+		parkedMessagesURL := fmt.Sprintf("/streams/$persistentsubscription-%s::%s-parked/head/backward/1", eventStreamID, groupName)
+		log.Info(parkedMessagesURL)
+		parkedMessagesResult := get(parkedMessagesURL, false)
+		eTagString, _ := jp.GetString((<-parkedMessagesResult).result, "eTag")
+		lastEventNumber, lastEventNumberErr := strconv.ParseInt(strings.Split(eTagString, ";")[0], 10, 64)
+		if lastEventNumberErr == nil {
+			lastEventNumber++ // +1 because Ids start from 0
+			log.Info(lastEventNumber)
+		}
+
+		// get $tb (truncate before) value from metadata
+		metadataURL := fmt.Sprintf("/streams/$persistentsubscription-%s::%s-parked/metadata", eventStreamID, groupName)
+		log.Info(metadataURL)
+		metadataResultChan := get(metadataURL, false)
+		metadataResult := <-metadataResultChan
+		tbValue, _ := jp.GetInt(metadataResult.result, "$tb")
+		log.Info("$tb Value: " + fmt.Sprint(tbValue))
+		totalNumberOfParkedMessages := float64(lastEventNumber - tbValue)
+		log.Info("Total number of parked messages:" + fmt.Sprint(totalNumberOfParkedMessages))
+
+		// get oldest message in the queue
+		getOldestMessageURL := fmt.Sprintf("/streams/$persistentsubscription-%s::%s-parked/%s/forward/1", eventStreamID, groupName, fmt.Sprint(lastEventNumber-int64(totalNumberOfParkedMessages)))
+		getOldestMessageResultChan := get(getOldestMessageURL, false)
+		getOldestMessageResult := <-getOldestMessageResultChan
+		oldestMessageUpdatedDate := ""
+		jsonparser.ArrayEach(getOldestMessageResult.result, func(value []byte, dataType jsonparser.ValueType, offset int, err error) {
+			oldestMessageUpdatedDate, _ = jp.GetString(value, "updated")
+		}, "entries")
+		log.Info("Oldest Message Date: " + oldestMessageUpdatedDate)
+
+		// add metric
+		ch <- prometheus.MustNewConstMetric(desc, prometheus.GaugeValue, totalNumberOfParkedMessages, eventStreamID, groupName, oldestMessageUpdatedDate)
```

Comment:
better convert to float64 at the moment of usage (here passing the totalNumberOfParkedMessages)